apps/testingaccessibility/src/pages/api/prices.ts

```diff
@@ -44,13 +44,13 @@ const pricesHandler = async (req: NextApiRequest, res: NextApiResponse) => {
             return upgradeProductIds.includes(purchase.productId)
           })?.id
 
-      const couponId = await getActiveCouponId({
-        siteCouponId,
-        coupon,
-        code,
-        productId,
-        spanContext,
-      })
+      const {activeMerchantCoupon, defaultCoupon} =
+        await getActiveMerchantCoupon({
+          siteCouponId,
+          code,
+          productId,
+          spanContext,
+        })
```

Comment:
instead of just the merchant coupon id we get the whole coupon as well as any backing (default) coupon that we can pass along. It only exists if it is valid and associated with the active merchant coupon.